src/chainlit/frontend/src/components/organisms/playground/index.tsx

```diff
@@ -37,20 +37,19 @@ export default function PromptPlayground() {
   const [providersError, setProvidersError] = useState();
   const [isDrawerOpen, toggleDrawer] = useToggle(false);
 
-  const isFirstRender = useIsFirstRender();
   const theme = useTheme();
   const isSmallScreen = useMediaQuery<Theme>((theme) =>
     theme.breakpoints.down('md')
   );
 
-  if (isFirstRender) {
+  useEffect(() => {
     client
       .getLLMProviders()
       .then((res) =>
         setPlayground((old) => ({ ...old, providers: res.providers }))
       )
       .catch((err) => setProvidersError(err));
-  }
+  }, [playground?.prompt?.provider]);
```

Comment:
why is this required? everytime we open the prompt playground the firstRender is triggered no?

Reply:
The first render happens before the modal is open, and the component returns `null` (see https://github.com/Chainlit/chainlit/pull/312/files#diff-96df05b06bc651bccb782e3ce335080f28b7cea2308758d45896652e3c2d1ccaR68-R70), this is happening when the page is displayed.

The provider is created by the langchain callback. This means the langchain chain must have started before we have the right providers available (instead of the default ones).
This isn't the perfect way to refresh the llm providers as it creates a couple extra requests, but it does solve our issue with minimal overhead.